### PR TITLE
docs(react-scheduler): add the TypeScript guide

### PR DESCRIPTION
**Pull request by MaximKudriavtsev**

### CODE REVIEW

**Comment by MaximKudriavtsev** [comment]
FPR

**Comment by MaximKudriavtsev** [comment]
@churkin all tests have been done 🎉

File: packages/dx-react-scheduler/docs/guides/typescript.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# React Scheduler - TypeScript
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/guides/typescript.md]:
```suggestion
# React Scheduler - TypeScript Support
```

File: packages/dx-react-scheduler/docs/guides/typescript.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+[TypeScript](https://www.typescriptlang.org) is a popular JavaScript superset with a strict syntax, developed by Microsoft. It supports static typing and enables catching errors at build time.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/guides/typescript.md]:
```suggestion

```

File: packages/dx-react-scheduler/docs/guides/typescript.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+The React Scheduler comes with TypeScript definitions that allows you to use it into TypeScript projects.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/guides/typescript.md]:
```suggestion
The React Scheduler is supplied with TypeScript definitions. Strict typing allows you to catch bugs at earlier stages and use features like code completion and automated refactoring.
```

File: site/_data/docs/navigation.yml [hunk 4]
```diff
@@ -323,6 +323,8 @@
         path: /react/scheduler/docs/guides/editing
       - title: Localization
         path: /react/scheduler/docs/guides/localization
+      - title: TypeScript
```
Comment (RomanTsukanov) [inline site/_data/docs/navigation.yml]:
```suggestion
      - title: TypeScript Support
```

**Pull request merged**